### ISSUE TITLE
Crash: autoFocus failed for some devices

### ISSUE THREAD

**Issue by dpeters-ipc** [dm77/barcodescanner#54]
I noticed several crashes in our crashlytics where autoFocus failed:

java.lang.RuntimeException: autoFocus failed
       at android.hardware.Camera.native_autoFocus(Camera.java)
       at android.hardware.Camera.autoFocus(Camera.java:1120)
       at me.dm7.barcodescanner.core.CameraPreview$1.run(SourceFile:203)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:212)
       at android.app.ActivityThread.main(ActivityThread.java:5137)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:718)
       at dalvik.system.NativeStart.main(NativeStart.java)

I already check if the device has autoFocus (hasSystemFeature(PackageManager.FEATURE_CAMERA_AUTOFOCUS)). If not, the Auto-Focus Button is disabled.

Affected devices:
- several Sony (see screenshot)
- Samsung Galaxy S2

![dm77_crashlytics_01](https://cloud.githubusercontent.com/assets/10939801/6126182/0e380dba-b11e-11e4-9d5b-d206acc5fcb0.png)

**Comment by Jakevin** [comment]
I have the same issue.

My device SONY Z ULTRA C6833.

```
java.lang.RuntimeException: autoFocus failed
       at android.hardware.Camera.native_autoFocus(Camera.java)
       at android.hardware.Camera.autoFocus(Camera.java:1120)
       at me.dm7.barcodescanner.core.CameraPreview$1.run(CameraPreview.java:203)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:212)
       at android.app.ActivityThread.main(ActivityThread.java:5135)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:877)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by dm77** [comment]
Merged in #41 might help. Thanks to @plastiv for the commit.

**Comment by dm77** [comment]
Don't think this is resolved, at least according to #85
